common/djangoapps/student/tests/tests.py

```diff
@@ -1145,3 +1169,133 @@ def _assert_responses(self, response, count):
         self.assertContains(response, 'This course is 1 of 3 courses in the', count)
         self.assertContains(response, self.program_name, count)
         self.assertContains(response, 'View XSeries Details', count)
+
+    def test_get_programs_method_without_cache(self):
+        """Verify that if programs config has cache disabled than
+        _get_course_programs method will not cache the response and will hit
+        the api for each request.
+        """
+        self.create_config(enabled=True, enable_student_dashboard=True, enable_cache=False, cache_ttl=100)
+        course_list = [u'edx/demox/Run_1', u'valid/edX/Course']
+        with patch('student.views.get_course_programs_for_dashboard') as mock_method:
+            mock_method.return_value = self.raw_data
+
+            # without enabling cache each time method will be called
+            parse_data1 = _get_course_programs(self.user, course_list)
+            mock_method.assert_called_once_with(self.user, course_list)
+            self.assertTrue(mock_method.called)
+            self.assert_parsed_data(parse_data1)
+
+            mock_method.reset_mock()
+
+            # method will be called again
+            parse_data2 = _get_course_programs(self.user, course_list)
+            mock_method.assert_called_once_with(self.user, course_list)
+            self.assertTrue(mock_method.called)
+            self.assert_parsed_data(parse_data2)
+
+    def test_get_programs_method_with_cache(self):
```

Comment:
```python
def test_get_programs_method_with_cache(self):
""" Verify that if the programs config has cache enabled than the
method '_get_course_programs' will cache the response and it will
not hit the Programs API until the programs cache remains available.
"""
```